### PR TITLE
bug: Empty sources and offset issue

### PR DESCRIPTION
**Pull request by j03m**
This PR corrects two bugs:

* For source maps, the sources property can potentially be an empty array if mappings
doesn't provide any information about its origin. For example if you compile an empty
js file with babel, a valid sourcemap can be created with `[]` for `sources`. Here
we add a fallback to `file` in this case and adds a fixture to test for it.

* When a file has dos line endings `\r\n` we found a few instances in a large codebase
where v8 would report the position of the `\n` as `startCol` for a range. This would
cause the starting column supplied to `sourceMap.originalPositionFor` to be a `-1`
because the startColumns of the transpiled source lines were calculated as
`endCol + length of line break sequence`. This change clamps the supplied value to
max 0 and adds a test fixture (force encoded as `\r\n`) to test for it.

### CODE REVIEW

File: test/v8-to-istanbul.js [hunk 1]
```diff
@@ -89,6 +90,49 @@ ${'//'}${'#'} sourceMappingURL=data:application/json;base64,${base64Sourcemap}
...
+      // (these ranges can occur on v8 running on windows) and verify
+      // coverage is applied correctly. CovLine will have a
+      // gap between the endCol of the previous line ending on \r and startCol of the
+      // next line. This would cause -1 to be sent for the source map lookup.
+      // This would cause source map translation to throw
+      v8ToIstanbul.applyCoverage([{
+        functionName: 'fake',
+        ranges: [{
+          startOffset: matchedNewLineChar - 1,
+          endOffset: matchedNewLineChar + 10
```
Comment (scottrobertwhittaker) [inline test/v8-to-istanbul.js:113]: Why 10?

**Review by scottrobertwhittaker — COMMENTED** [review]
I think the substantive change here looks good. My only suggestion would be splitting the two fixed bugs into two separate commits (along with maybe separate ones for your `.gitignore` and `standard` changes?)

File: test/v8-to-istanbul.js (hunk 1 above)
Comment (j03m) [reply test/v8-to-istanbul.js:113]: No real reason. I just needed a range. Only the first char position is important for the bug to manifest.

**Comment by j03m** [comment]
Thanks @scottrobertwhittaker. I think I may still submit it as one PR since its a low change/volume repo and see what they say.

**Comment by j03m** [comment]
clsing will issue upstream

**Comment by scottrobertwhittaker** [comment]
Thanks! I didn't mean two PRs, just two commits in the same PR

**Pull request closed without merging**